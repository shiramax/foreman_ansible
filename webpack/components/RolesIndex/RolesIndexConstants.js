import React from 'react';

export const ANSIBLE_ROLES_API = '/ansible/api/v2/ansible_roles/';
export const linkCellFormatter = ({ title: url }) => ({
  children: <a href={url.split(' ')[1]}>{url.split(' ')[0]}</a>,
});
