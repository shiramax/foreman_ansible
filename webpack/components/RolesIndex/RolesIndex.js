import React from 'react';
import { Table, TableHeader, TableBody } from '@patternfly/react-table';
import PropTypes from 'prop-types';
import { linkCellFormatter } from './RolesIndexConstants';
import withPagination from '../WithPagination';

const RolesIndex = ({ rows, RolesIndexActions, paginatedRows }) => {
  const cols = [
    { title: 'Name' },
    { title: 'Variables', cellTransforms: [linkCellFormatter] },
    { title: 'Hostgroups' },
    { title: 'Hosts', cellTransforms: [linkCellFormatter] },
    { title: 'Imported at' },
  ];

  return (
    <div id="roles-index-table">
      <Table
        aria-label="roles index"
        rows={paginatedRows}
        actions={RolesIndexActions(rows)}
        cells={cols}
      >
        <TableHeader />
        <TableBody />
      </Table>
    </div>
  );
};

RolesIndex.defaultProps = {
  rows: [],
  RolesIndexActions: () => {},
  paginatedRows: [],
};

RolesIndex.propTypes = {
  rows: PropTypes.array,
  RolesIndexActions: PropTypes.func,
  paginatedRows: PropTypes.array,
};
export default withPagination(RolesIndex);
