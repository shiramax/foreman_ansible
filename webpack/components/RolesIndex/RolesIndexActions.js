import { APIActions } from 'foremanReact/redux/API';
import { translate as __ } from 'foremanReact/common/I18n';
import { ANSIBLE_ROLES_API } from './RolesIndexConstants';

export const foremanUrl = path => `${window.URL_PREFIX}${path}`;

export const onDelete = role =>
  APIActions.delete({
    key: 'delete_ansible_role',
    url: `${ANSIBLE_ROLES_API}${role.id}`,
    handleSuccess: () => window.location.reload(),
    successToast: response => `Role ${role.name} was successfully deleted`,
    errorToast: error =>
      `${__('Failed to import roles and variables ')} ${error}`,
  });
