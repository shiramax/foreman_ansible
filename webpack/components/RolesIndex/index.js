import React from 'react';
import { useDispatch } from 'react-redux';

import RolesIndex from './RolesIndex';
import { onDelete } from './RolesIndexActions';

const WrappedRolesIndex = props => {
  const dispatch = useDispatch();
  const actions = rows => [
    {
      title: 'Delete',
      onClick: (event, rowId, rowData, extra) => {
        event.preventDefault();
        dispatch(onDelete(rows[rowId].role));
      },
    },
  ];

  return <RolesIndex RolesIndexActions={actions} {...props} />;
};

export default WrappedRolesIndex;
