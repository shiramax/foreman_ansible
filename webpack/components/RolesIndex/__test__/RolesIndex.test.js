import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import RolesIndex from '../RolesIndex';

const rowsData = [
  {
    cells: [
      'dcos.cluster.dcos_agent',
      '1 /ansible/ansible_variables?search=ansible_role+%3D+dcos.cluster.dcos_agent',
      0,
      '0 /hosts?search=ansible_role+%3D+dcos.cluster.dcos_agent',
      'about 17 hours ago',
    ],

    id: 'dcos.cluster.dcos_agent',
    can_delete: true,
  },
  {
    cells: [
      'dcos.cluster.dcos_bootstrap',
      '8 /ansible/ansible_variables?search=ansible_role+%3D+dcos.cluster.dcos_bootstrap',
      0,
      '0 /hosts?search=ansible_role+%3D+dcos.cluster.dcos_bootstrap',
      'about 17 hours ago',
    ],
    id: 'dcos.cluster.dcos_bootstrap',
    can_delete: false,
  },
];

const fixtures = {
  'should render': {
    rowsData,
    RolesIndexactions: () => {},
  },
};

describe('RolesIndexTable', () =>
  testComponentSnapshotsWithFixtures(RolesIndex, fixtures));
