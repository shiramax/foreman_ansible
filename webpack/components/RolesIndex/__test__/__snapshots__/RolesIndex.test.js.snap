// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RolesIndexTable should render 1`] = `
<div>
  <RolesIndex
    RolesIndexActions={[Function]}
    paginatedRows={
      Array [
        Object {
          "can_delete": true,
          "cells": Array [
            "dcos.cluster.dcos_agent",
            "1 /ansible/ansible_variables?search=ansible_role+%3D+dcos.cluster.dcos_agent",
            0,
            "0 /hosts?search=ansible_role+%3D+dcos.cluster.dcos_agent",
            "about 17 hours ago",
          ],
          "id": "dcos.cluster.dcos_agent",
        },
        Object {
          "can_delete": false,
          "cells": Array [
            "dcos.cluster.dcos_bootstrap",
            "8 /ansible/ansible_variables?search=ansible_role+%3D+dcos.cluster.dcos_bootstrap",
            0,
            "0 /hosts?search=ansible_role+%3D+dcos.cluster.dcos_bootstrap",
            "about 17 hours ago",
          ],
          "id": "dcos.cluster.dcos_bootstrap",
        },
      ]
    }
    rows={
      Array [
        Object {
          "can_delete": true,
          "cells": Array [
            "dcos.cluster.dcos_agent",
            "1 /ansible/ansible_variables?search=ansible_role+%3D+dcos.cluster.dcos_agent",
            0,
            "0 /hosts?search=ansible_role+%3D+dcos.cluster.dcos_agent",
            "about 17 hours ago",
          ],
          "id": "dcos.cluster.dcos_agent",
        },
        Object {
          "can_delete": false,
          "cells": Array [
            "dcos.cluster.dcos_bootstrap",
            "8 /ansible/ansible_variables?search=ansible_role+%3D+dcos.cluster.dcos_bootstrap",
            0,
            "0 /hosts?search=ansible_role+%3D+dcos.cluster.dcos_bootstrap",
            "about 17 hours ago",
          ],
          "id": "dcos.cluster.dcos_bootstrap",
        },
      ]
    }
  />
  <Pagination
    className=""
    defaultToFullPage={true}
    firstPage={1}
    isCompact={true}
    isDisabled={false}
    isSticky={false}
    itemCount={2}
    itemsEnd={null}
    itemsStart={null}
    offset={0}
    onFirstClick={[Function]}
    onLastClick={[Function]}
    onNextClick={[Function]}
    onPageInput={[Function]}
    onPerPageSelect={[Function]}
    onPreviousClick={[Function]}
    onSetPage={[Function]}
    ouiaSafe={true}
    page={0}
    perPage={20}
    perPageOptions={
      Array [
        Object {
          "title": "5",
          "value": 5,
        },
        Object {
          "title": "10",
          "value": 10,
        },
        Object {
          "title": "20",
          "value": 20,
        },
        Object {
          "title": "50",
          "value": 50,
        },
      ]
    }
    titles={
      Object {
        "paginationTitle": "top pagination",
      }
    }
    toggleTemplate={[Function]}
    variant="top"
    widgetId="pagination-options-menu"
  />
</div>
`;
